pkg/arvo/sys/hoon.hoon

```diff
@@ -11534,114 +11557,175 @@
   ++  glom
     |=  [wit=whit taw=whit]
     ^-  whit
-    :*  ?~(lab.wit lab.taw lab.wit)
-        ?~(boy.wit boy.taw boy.wit)
-        (~(uni by def.wit) def.taw)
+    :*  (~(uni by bat.wit) bat.taw)
         (~(uni in use.wit) use.taw)
     ==
   ++  docs
     |%
+    ::  +apex: prefix comment. may contain batch comments.
     ::
-    ::  above core
-    ::
+    ::    when a prefix doccord is parsed, it is possible that there is no +gap
+    ::    afterward to be consumed, so we add an additional newline and set back
+    ::    the line of the n
     ++  apex
+      %+  knee  *whit  |.  ~+
       ;~  plug
-        =/  ron  (punt (indo noel))
-        (punt (ifix [ron ron] (into head)))             ::  label
-      ::
-        =/  ron  (punt (indo null))
-        (ifix [ron ron] (punt body))                    ::  body
+        |=  tub=nail
+        =/  vex
+          %.  tub
+          %-  star  %-  myth
+          ;~(pfix (punt leap) (star ace) col col ;~(pose larg smol))
+        ?~  q.vex  vex
+        :-  p=p.vex
+        %-  some
+        ?~  p.u.q.vex
+          [p=~ q=q.u.q.vex]
+        :-  p=(malt p.u.q.vex)
+        q=`nail`[[(dec p.p.q.u.q.vex) q.p.q.u.q.vex] ['\0a' q.q.u.q.vex]]
       ::
-        (cook malt (star fill))                         ::  definitions
-        (easy ~)                                        ::  defs used (none)
+        (easy ~)
       ==
     ::
-    ::  backward line
-    ::
+    ::  +apse: postfix comment.
     ++  apse
+      %+  knee  *help  |.  ~+
       ;~  pose
-        %+  cook  |=([a=term b=cord] %*(. *whit def (my [a b ~] ~)))
-        (exit fine)
-      ::
-        %+  cook  |=(a=cord %*(. *whit boy `[a ~]))
+        %+  cook  |=(a=cord %*(. *help summary.crib a))
+        ::TODO: if there is a $link then interpret it as the start
+        ::of a prefix comment instead of postfix
```

Comment:
Yeah, but I'm rethinking why and I think I just couldn't figure out the right way to implement it and punted the question til now.

```hoon
|=  ::  .foo: a pat
     ::  .bar: a tar
     [foo=@ bar=*]
[foo bar]
```

which I'd like to attach to the definitions below. The way it is currently written, this could not happen.

So there's two ways to do it - always interpret comments on the same line as postfix comments, so `.foo: a pat` will initially try to attach to `|=`, then it will need to be moved later to `.foo` via some kind of search process. Or say that postfix comments can't have a `$link` and so it must be a prefix comment, and then it will attach to the sample structure via the usual process. The latter sounds easier to understand to me and to implement, but the former is probably how it will look when `(list link)`s are processed anyways and is probably a lot more work to do correctly.

I didn't implement either method yet though, and its at least clear to me that the above comment style really ought to work, and its suprising that it doesn't, so it violates the principle of least surprise.. I'm leaning towards the latter way even though it will have to be ripped out later, and building the former might be grant material.